### PR TITLE
[Tools] Add editor snippets to help new developers.

### PR DESCRIPTION
**Pull request by mandel-macaque**
We do have a few tings that we have to do manually yet it is repetitive,
I am adding the vscode (soon to come vim) configuration files to provide
useful snippets when working with the bindings.

The syntax of the snippets present is very simple, there are just two
things to understand in these new ones:

1. Tab completions: $1, $2 etc.. are the tab complitions that the user
  can provide. This allows the user to navigate and the snippets engine
  will replace all the complitions with the provided text.
2. Options: We can provide options for tab complitions, for example for
  the PlatformName, which is a known value. The syntax to do so is
  {$1|one,two,three|} where:

  * `$1`: is the tab completion.
  * `|one,two,three|`: are the options.
 

https://user-images.githubusercontent.com/2190086/129991926-520d0297-ac23-4f05-b9d2-2aa07ec14e25.mov

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: tools/editors/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## VS Code
+
+You can find the snippets for csharp in the VSCode directory. There are two possible scenarios:
+
+1. You do no have snippets setup in your system.
+2. You do alrady use snippets.
+
+### New to snippets 
+
+In order to be able to use snippets you need to enable tab completion on vscode, do yo so ny following the instructions
```
Comment (rolfbjarne) [inline tools/editors/README.md:15]:
```suggestion
In order to be able to use snippets you need to enable tab completion on vscode, you do so by following the instructions 
```

I think?

File: tools/editors/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Copy the snippets file
+
+```bash
+mkdir -p "~/Library/Application Support/Code/User/snippets" 
+cp VSCode/* "~/Library/Application Support/Code/User/snippets" 
+```
+
+### Already have snippets
+
+You are al setup, you just need to merge the contents of the file VSCode/csharp.json with the file in "~/Library/Application Support/Code/User/snippets"
```
Comment (rolfbjarne) [inline tools/editors/README.md:27]:
```suggestion
You are all setup, you just need to merge the contents of the file VSCode/csharp.json with the file in "~/Library/Application Support/Code/User/snippets" 
```

**Review by rolfbjarne — APPROVED** [review]

File: tools/editors/VSCode/csharp.json [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		],
+		"description": "Add a native ref type declaration"
+	},
+
+	"Import needed dlls for versioning.": {
+		"prefix": "netcore_versioning",
+		"body": [
+		  "#if NET",
+		  "using System.Runtime.Versioning;",
+		  "#endif"
```
Comment (spouliot) [inline tools/editors/VSCode/csharp.json:60]: minor: `System.Runtime.Versioning` exists in the legacy BCL so the `#if NET` wrapper is not needed

**Review by spouliot — APPROVED** [review]

File: tools/editors/VSCode/csharp.json (hunk 3 above)
Comment (mandel-macaque) [reply tools/editors/VSCode/csharp.json:60]: Simple fix. The ones I wrote for vim are much nicer, I'm doing visual replacements, so we can select text and add the needed stuff, I'll do those in a diff pr

Comment (mandel-macaque) [inline tools/editors/VSCode/csharp.json:60]:
```suggestion
		  "using System.Runtime.Versioning;"
```

**Pull request merged**